pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -264,18 +269,24 @@ func (r *reconciler) reconcile(ctx context.Context, mr *resourcesv1alpha1.Manage
 					continue
 				}
 
+				hash.Write(value)
```

Comment:
Be careful here: maps are unsorted in go. Hence, the checksum that you calculate here will be unstable.

Please add a test case covering this (multiple secret keys) and one case for multiple secrets.

Reply:
@ary1992 this is not resolved. Can you take care of it or have you checked that my comment is not relevant?

Reply:
While traversing the map with for loop, keys appear in the same order as they are stored on the map. So in the case of secrets with multiple data keys, the keys will appear in the same order as they were stored initially. 
Moreover, to ensure the same order of keys we need to either traverse the secret's data in sorted order of keys or we can store checksum of each entry in secret's data(this will be an overhead ).

Reply:
I don't understand your comment. There is no "initial" or deterministic order of keys in `secret.Data`, as maps are not sorted in go.
Please see https://yourbasic.org/golang/sort-map-keys-values/ for a coding example of how to solve this problem for your case.

Reply:
Yeah, I thought of this as a solution. But then I thought it will be an overhead to sort each secret's data keys and then traverse it.

Reply:
Done.